services/src/main/java/org/keycloak/protocol/oidc/grants/ciba/channel/AuthenticationRequest.java

```diff
@@ -0,0 +1,181 @@
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+package org.keycloak.protocol.oidc.grants.ciba.channel;
+
+import javax.crypto.SecretKey;
+import java.io.UnsupportedEncodingException;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.keycloak.OAuth2Constants;
+import org.keycloak.crypto.Algorithm;
+import org.keycloak.crypto.KeyUse;
+import org.keycloak.crypto.SignatureProvider;
+import org.keycloak.crypto.SignatureSignerContext;
+import org.keycloak.jose.jwe.JWEException;
+import org.keycloak.jose.jws.JWSBuilder;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.protocol.oidc.grants.ciba.CibaGrantType;
+import org.keycloak.representations.IDToken;
+import org.keycloak.representations.JsonWebToken;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.keycloak.services.Urls;
+import org.keycloak.util.TokenUtil;
+
+/**
+ * <p>Represents an authentication request sent by a consumption device.
+ *
+ * <p>A authentication request can be serialized to a JWE so that they can be exchanged with authentication devices to communicate
+ * and authorize the authentication request made by consumption devices.
+ */
+public class AuthenticationRequest extends JsonWebToken {
+
+    /**
+     * Deserialize the given {@code jwe} to a {@link AuthenticationRequest} instance.
+     *
+     * @param session the session
+     * @param jwe the authentication request in JWE format.
+     * @return the authentication request instance
+     * @throws Exception
+     */
+    public static AuthenticationRequest deserialize(KeycloakSession session, String jwe) {
+        SecretKey aesKey = session.keys().getActiveKey(session.getContext().getRealm(), KeyUse.ENC, Algorithm.AES).getSecretKey();
+        SecretKey hmacKey = session.keys().getActiveKey(session.getContext().getRealm(), KeyUse.SIG, Algorithm.HS256).getSecretKey();
+
+        try {
+            byte[] contentBytes = TokenUtil.jweDirectVerifyAndDecode(aesKey, hmacKey, jwe);
+            jwe = new String(contentBytes, "UTF-8");
+        } catch (JWEException | UnsupportedEncodingException e) {
+            throw new RuntimeException("Error decoding auth_req_id.", e);
+        }
+
+        return session.tokens().decode(jwe, AuthenticationRequest.class);
+    }
+
+    public static final String SESSION_STATE = IDToken.SESSION_STATE;
+    public static final String AUTH_RESULT_ID = "auth_result_id";
+
+    @JsonProperty(OAuth2Constants.SCOPE)
+    protected String scope;
+
+    @JsonProperty(SESSION_STATE)
+    protected String sessionState;
```

Comment:
The "sessionState" is not used at all in the AuthenticationRequest. It is always null. Should we remove this parameter?

Reply:
@mposolda I agree with you.